### PR TITLE
Add support for new paste function

### PR DESCRIPTION
**Pull request by hanford**
This PR adds support for broadcasting `paste` events that slate-react properly receives and handles to various `insertData` functions.

My test looks like this:
```ts
import { assertOutput, buildTestHarness } from 'slate-test-utils';
import { jsx as h } from 'slate-hyperscript';

import TextEditor from '..';

it('user copy and pastes link', async () => {
  const input = h('editor', [h('element', { type: 'paragraph' }, [h('text', [h('cursor')])])]);

  const [editor, { paste }] = await buildTestHarness(TextEditor)({
    editor: input,
  });

  await paste('foo!');

  assertOutput(
    editor,
    h('editor', [h('element', { type: 'paragraph' }, [h('text', ['foo!', h('cursor')])])]),
  );
});
```

image

### CODE REVIEW

File: src/buildTestHarness.tsx [hunk 1]
```diff
@@ -142,6 +143,25 @@ export const buildTestHarness =
...
+      act(async () => {
+        const event = new window.Event('paste', {
+          bubbles: true,
+          cancelable: true,
+          composed: true,
+        })
+
+        // @ts-ignore Typescript doesn't expect clipboardData on Event type
+        event.clipboardData = {
+          types: ['text/html'],
```
Comment (mwood23) [inline src/buildTestHarness.tsx:156]: I'm not as familiar with this API, but what do you think about adding a second param to this function that that a `type`?

https://w3c.github.io/clipboard-apis/#reading-from-clipboard

```ts
type ClipboardDataType = 'text/html' | 'text/plain' | 'image/png'
const paste = async (payload: string, {types}: {types?: ClipboardDataType[] = ['text/html']}) =>
```

**Review by mwood23 — APPROVED** [review]
Awesome addition to harness, thank you for taking the time to put in the PR! One comment on the API that I think could be useful that I'm curious to get your thoughts on before merging.

File: src/buildTestHarness.tsx (hunk 1 above)
Comment (hanford) [reply src/buildTestHarness.tsx]: Updated!

**Comment by hanford** [comment]
Added an optional _option_ argument that will allow consumers to change the types value. Let me know if there is anything else!

**Review by mwood23 — APPROVED** [review]
Beautiful!

**Pull request merged**